### PR TITLE
pscanrules: Timestamp Disclosure include header field as param

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Add change note.
- TimestampDisclosureScanRule > Updated to include the header name as the alert parameter when applicable.
- TimestampDisclosureScanRuleUnitTest > Add/update tests to assert the new alert content.

## Related Issues
Fixes zaproxy/zaproxy#8160

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Comment by thc202** [comment]
Has conflicts now.

**Comment by kingthorin** [comment]
Fixed

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java [hunk 1]
```diff
@@ -138,28 +140,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
...
-                        .append(responseheader.getValue());
-            }
-        }
-
-        String responsebody = msg.getResponseBody().toString();
-        String[] responseparts = {filteredResponseheaders.toString(), responsebody};
+        List<HttpHeaderField> responseparts = new ArrayList<>();
+        responseparts.addAll(
+                msg.getResponseHeader().getHeaders().stream()
+                        .filter(header -> !RESPONSE_HEADERS_TO_IGNORE.contains(header.getName()))
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java:146]: Needs a test for the headers in `RESPONSE_HEADERS_TO_IGNORE` but with different case.

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java [hunk 2]
```diff
@@ -138,28 +140,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
...
+                        .collect(Collectors.toList()));
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java:147]: `.forEach(responseparts::add)`

**Comment by thc202** [comment]
The test `shouldNotRaiseAlertOnIgnorableHeaders` is broken as it does not pass the range checks. (Happy to fix it just noting.)

**Comment by kingthorin** [comment]
> The test `shouldNotRaiseAlertOnIgnorableHeaders` is broken as it does not pass the range checks. (Happy to fix it just noting.)

Fix coming, needed to add one for case insensitivity anyway so applied the same fix.

**Comment by kingthorin** [comment]
Review addressed.

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
Mind if I include a few other changes as a "clean code" move?

Merging the conditions at 173/174, removing `msg` from `getExtraInfo` as it isn't used, and dropping all the kinda useless/boilerplate javadoc?

**Comment by thc202** [comment]
A separate commit sounds good.

**Comment by kingthorin** [comment]
Added

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java [hunk 3]
```diff
@@ -221,41 +203,19 @@ public int getRisk() {
...
 
-    /**
-     * gets extra information associated with the alert
-     *
-     * @param msg
-     * @param arg0
-     * @return
-     */
-    private String getExtraInfo(HttpMessage msg, String evidence, Date timestamp) {
+    private String getExtraInfo(String evidence, Date timestamp) {
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java:218]: Could be made static also.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you again!

**Review by psiinon — APPROVED** [review]

**Pull request merged**